instrumentation/opentelemetry-instrumentation-httpx/src/opentelemetry/instrumentation/httpx/__init__.py

```diff
@@ -648,6 +649,15 @@ def __init__(self, *args, **kwargs):
             request_hook=_InstrumentedClient._request_hook,
             response_hook=_InstrumentedClient._response_hook,
         )
+        if self._mounts:
+            for url_pattern, transport in self._mounts.items():
```

Comment:
Question: shouldn't we loop over the original mounts instead of changing the dict we are looping over?

Reply:
Indeed, looping over the dict while changing it isn't a good idea. Thanks for taking a look on this